onnxruntime/core/providers/cuda/nn/pool.cc

```diff
@@ -180,22 +180,17 @@ Status Pool<T, PoolType, NHWC>::ComputeInternal(OpKernelContext* context) const
   if (kernel_shape.size() < 2) {
     // cudnn only takes 4D or 5D input, so pad dimensions if needed
     if (NHWC) {
-      x_dims_cudnn.insert(x_dims_cudnn.begin() + 1, 1);
-      y_dims_cudnn.insert(y_dims_cudnn.begin() + 1, 1);
-      ORT_ENFORCE(pads.size() >= kernel_shape.size());
-      pads.insert(pads.begin() + kernel_shape.size(), 0);
-      pads.insert(pads.end(), 0);
-      kernel_shape.insert(kernel_shape.begin() + 1, 1);
-      strides.insert(strides.begin() + 1, 1);
+      x_dims_cudnn.insert(x_dims_cudnn.begin() + 2, 1);
```

Comment:
Is there a plan to add 1-D NHWC test(s) to keep this path covered ?
    * If not, maybe we can just return back NOT_IMPLEMENTED status back to the user for now ? 1-D Pool is generally rare and we can just add the enhancement later with tests ? 

And how about adding a test for the 1-D NCHW version with all padding not zeros (as a test to cover this regression fix) ?

Reply:
I don't know how to write test cases for the new NHWC EP. I think we need to feed in different data based on whether the EP is NHWC or NCHW. Is there any example with OpTester?  If not, yes I can disable this code path.   @tianleiwu , what do you think?

Reply:
You can remove NHWC from the list of excluded providers, then it will test NHWC when ENABLE_CUDA_NHWC_OPS is defined (like in current Linux GPU pipeline).

Currently it is excluded for Conv tests:
https://github.com/microsoft/onnxruntime/blob/89aa4697b145b3879c8a0f40db353af1e5e5003a/onnxruntime/test/providers/cpu/nn/conv_op_test.cc#L62-L63

Reply:
@snnn - I think you can try adding a 1-D test like this (https://github.com/microsoft/onnxruntime/blob/89aa4697b145b3879c8a0f40db353af1e5e5003a/onnxruntime/test/providers/cuda/nhwc/pool_test.cc#L53) - the padding and strides need to be adjusted for the 1-D case (it is defaulted to the 2-D case). As long as that passes, it should be good as the way it operates is to run the op in NHWC mode and compare the results with the NCHW mode of the operator.

Reply:
No, the NHWC CUDA code for 1D totally doesn't work. 

Pool::ComputeInternal function has:
```
auto out_channel = NHWC ? x_shape[3] : x_shape[1];
```
Here it assumes the x_shape at least has 4 dims.
So I will disable this code path for now.

Reply:
Yep - makes sense. It looks like we need fixes in more places to account for the 1-D case